go/kbfs/libkbfs/favorites_test.go

```diff
@@ -163,23 +171,30 @@ func TestFavoritesAddRemoveAdd(t *testing.T) {
 		Data:    favorites.Data{},
 		Created: false,
 	}
-	config.mockKbpki.EXPECT().FavoriteList(gomock.Any()).Return(keybase1.FavoritesResult{}, nil)
 	folder1 := fav1.ToKBFolder()
 
 	config.mockKbpki.EXPECT().FavoriteAdd(gomock.Any(), folder1).
-		Times(2).Return(nil)
-	config.mockKbpki.EXPECT().FavoriteDelete(gomock.Any(), folder1).
 		Return(nil)
-
+	config.mockKbpki.EXPECT().FavoriteList(gomock.Any()).Return(keybase1.FavoritesResult{
+		FavoriteFolders: []keybase1.Folder{folder1},
+	}, nil)
 	if err := f.Add(ctx, fav1); err != nil {
 		t.Fatalf("Couldn't add favorite: %v", err)
 	}
 
+	config.mockKbpki.EXPECT().FavoriteDelete(gomock.Any(), folder1).
+		Return(nil)
+	config.mockKbpki.EXPECT().FavoriteList(gomock.Any()).Return(keybase1.FavoritesResult{}, nil)
 	if err := f.Delete(ctx, fav1.Folder); err != nil {
 		t.Fatalf("Couldn't delete favorite: %v", err)
 	}
 
-	// A second add shouldn't result in a KBPKI call
```

Comment:
I couldn't think of a reason why the second add shouldn't result in a KBPKI call. That sounds like a bug we had. So I got rid of this comment and added an `FavoriteAdd`. But shout if anyone thinks I'm wrong here!

Reply:
YOU'RE WRONG!  (Sorry, you asked for a shout.)  For the reasons I outlined above -- we don't want to spam the server with multiple identical add calls.  Why does your code change this though, the check for not duplicating add calls is still there as far as I can tell, right?

Reply:
haha thanks. I actually don't know why it used to pass. Note that there's a Delete right above this, on the same folder, which also went to the API server. So this needs another RPC to add it back, right?

Reply:
Oohhh, sorry, I just searched for "// A second add" in `favorites_test.go` and was looking at the first result in `TestFavoritesAddTwice`, so was responding based on that.

In this case there's already a `Times(2)` on the add call above -- I guess the comment was just copy and pasted, and wrong. So with your code, it's being called 3 times now?  That seems incorrect too.

Reply:
Got it. That makes sense. That `.Times(2)` is removed actually, so sounds like this is correct as-in!

Reply:
(I think I removed that `Times(2)` at some point when fixing the tests but forgot about that when I made the comment here, which is why I thought we had a bug)

Reply:
Ah got it.